src/integrations/watchers/indexable-post-watcher.php

```diff
@@ -228,6 +219,32 @@ protected function update_has_public_posts( $indexable ) {
 		}
 	}
 
+	/**
+	 * Checks if given attachment has any public posts.
+	 *
+	 * @param int       $parent_id The parent post id.
+	 * @param Indexable $indexable The indexable for the attachment.
+	 *
+	 * @return bool True when has public parent indexable.
+	 */
+	protected function attachment_has_public_posts( $parent_id, Indexable $indexable ) {
+		if ( $parent_id === 0 ) {
+			return false;
+		}
+
+		if ( $indexable->post_status !== 'inherit' ) {
+			return false;
+		}
+
+		try {
+			$post_parent_indexable = $this->repository->find_by_id_and_type( $parent_id, 'post' );
+		} catch ( Exception $exception ) { // @codingStandardsIgnoreLine Generic.CodeAnalysis.EmptyStatement.DetectedCATCH -- There is nothing to do.
+			return false;
```

Comment:
I think this CS ignore is not necessary, since the `catch` statement is not empty.